### PR TITLE
Saga id generator

### PR DESCRIPTION
**Pull request by danielmarbach**

### CODE REVIEW

**Comment by danielmarbach** [comment]
@Scooletz not sure what you mean. I did it as a dedicated commit

**Comment by Scooletz** [comment]
Let me rephrase. This renamings could easily be put in a separate PR or directly to the `develop`. The crux of this PR is the saga refactor. This would make this PR much smaller - easier to read.

**Comment by danielmarbach** [comment]
@Scooletz fair enough

File: src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/When_persisting_a_saga_with_no_defined_unique_property.cs [hunk 1]
```diff
@@ -12,18 +12,16 @@ public class When_persisting_a_saga_with_no_defined_unique_property : SagaPersis
...
             configuration.RequiresFindersSupport();
 
-            var sagaId = Guid.NewGuid();
-            var sagaData = new SagaWithoutCorrelationPropertyData
-            {
-                Id = sagaId,
-                FoundByFinderProperty = sagaId.ToString()
-            };
+            var correlationPropertyData = Guid.NewGuid().ToString();
+            var sagaData = new SagaWithoutCorrelationPropertyData { FoundByFinderProperty = correlationPropertyData };
```
Comment (SeanFeldman) [inline src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/When_persisting_a_saga_with_no_defined_unique_property.cs]: A bit confusing that a saga w/o correlation property is using a data that is stored with a variable named `correlationPropertyData`

File: src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/When_completing_a_saga_with_no_defined_correlation_property.cs [hunk 2]
```diff
@@ -17,14 +17,16 @@ public class When_completing_a_saga_with_no_defined_correlation_property : SagaP
         {
             configuration.RequiresFindersSupport();
 
-            var sagaId = Guid.NewGuid();
+            var correlationPropertyData = Guid.NewGuid().ToString();
```
Comment (SeanFeldman) [inline src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/When_completing_a_saga_with_no_defined_correlation_property.cs]: If the test is for the no correlation property scenario, wouldn't `correlationPropertyData` be called something else? F.e. `propertyFoundByData`?

**Comment by danielmarbach** [comment]
@SeanFeldman addressed your comments

File: src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/DefaultSagaIdGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿namespace NServiceBus.Persistence.ComponentTests
+{
+    using System;
+    using Sagas;
+
+    class DefaultSagaIdGenerator : ISagaIdGenerator
+    {
+        public Guid Generate(SagaIdGeneratorContext context)
+        {
+            return CombGuid.Generate();
```
Comment (tmasternak) [inline src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/DefaultSagaIdGenerator.cs]: Can we just have `Guid.New()`?

File: src/NServiceBus.Persistence.ServiceFabric/SagaPersister/SagaPersister.cs [hunk 4]
```diff
@@ -28,13 +24,6 @@ public Task Complete(IContainSagaData sagaData, SynchronizedStorageSession sessi
                 {
```
Comment (tmasternak) [inline src/NServiceBus.Persistence.ServiceFabric/SagaPersister/SagaPersister.cs]: Do we need the second part of the condition here?

**Comment by tmasternak** [comment]
Is it worth opening another PR trying to reduce duplication in saga component tests?

**Comment by SeanFeldman** [comment]
Personally, I favor duplication in tests for clarity over complication for the sake of code duplication removal. My 2c.

**Comment by danielmarbach** [comment]
@tmasternak I do agree with @SeanFeldman Right now it is easier to understand those tests with the redundancy

File: src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/DefaultSagaIdGenerator.cs (hunk 3 above)
Comment (danielmarbach) [inline src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/DefaultSagaIdGenerator.cs]: I copied the core default persister since it is internal. If the tests would be run against persister which assumes core behavior it might be dangerous to use `Guid.NewGuid`

File: src/NServiceBus.Persistence.ServiceFabric/SagaPersister/SagaPersister.cs (hunk 4 above)
Comment (SeanFeldman) [inline src/NServiceBus.Persistence.ServiceFabric/SagaPersister/SagaPersister.cs]: The 2nd part checks if there's a concurrent process that has completed that same saga. Removing it would not change the course of things, but would hide the contention on the saga data.

Comment (danielmarbach) [inline src/NServiceBus.Persistence.ServiceFabric/SagaPersister/SagaPersister.cs]: Pretty sure we do, if the entry was modified from the point we last saw it we need to retry
